Vpn.Service/ManagerRpcService.cs

```diff
@@ -112,17 +135,28 @@ private async Task HandleRpcClientAsync(NamedPipeServerStream pipeServer, Cancel
         }
     }
 
-    private void RpcClientContinuation(Task task)
-    {
-        if (task.IsFaulted)
-            _logger.LogWarning(task.Exception, "Client RPC task faulted");
-        _activeClientTasks.TryRemove(task.Id, out _);
-    }
-
     private async Task HandleRpcMessageAsync(ReplyableRpcMessage<ServiceMessage, ClientMessage> message,
         CancellationToken ct)
     {
         _logger.LogInformation("Received RPC message: {Message}", message.Message);
         await _manager.HandleClientRpcMessage(message, ct);
     }
+
+    public async Task BroadcastAsync(ServiceMessage message, CancellationToken ct)
+    {
+        foreach (var (clientId, client) in _activeClients)
```

Comment:
Is there anything that prevents `_activeClients` from being modified while you are looping over it?

Reply:
Since it's a concurrent dictionary it's exception-safe. As far as I understand, any added or removed items from the dictionary while we're looping may or may not get called, but I think that's fine.